mapbox/dr-ui
# [Info arch] Add option to override <OnThisPage> default visibilty

This PR adds a new frontmatter property for layouts: `onThisPage`. If left unspecified, the default aside behavior will happen: aside will appear on pages with `layout: page`, but will not appear for any other layout. With this addition, `onThisPage: true` will force an aside to appear on pages with other layouts. `onThisPage: false` will cause asides to be hidden for pages with `layout: page`.

## How to test

In the catalog site, edit two pages: 

- For one of the guides with `layout: page`, add the following frontmatter: `onThisPage: false`. 
- For one of the examples, add the following frontmatter: `onThisPage: true`.
- Open the catalog site and verify that the first page you edited no longer contains an aside and that the second does!

## QA checklist



- [x] Accessibility score in [Chrome DevTools Audit/Lighthouse](https://developers.google.com/web/tools/lighthouse#devtools) is 98% with Lighthouse version: `6.2.0`.

Open the test cases app locally on:
